### PR TITLE
Only allow Administrator to assign roles to users

### PR DESCRIPTION
**Pull request by ShaydeNofziger**
Includes two new middleware:
* validateAuthenticationAttachUser, which attaches the user model for the requesting user on the req object.
* verifyAdministrator, which uses the req.user object attached via the above middleware to check whether the user is an admin, and 403's if they are not.

---
This change is [Reviewable](https://reviewable.io/reviews/andymeneely/dev-fortress-server/9)

### CODE REVIEW

File: test/.eslintrc.json [hunk 1]
```diff
@@ -6,6 +6,9 @@
     },
     "rules": {
       // Override default settings for this directory
```
Comment (jcdesimp) [inline test/.eslintrc.json]: Should have noticed this in the last review but commends aren't valid JSON

File: test/.eslintrc.json [hunk 2]
```diff
@@ -6,6 +6,9 @@
...
-      "no-underscore-dangle": "off"
+      "no-underscore-dangle": "off",
+      "no-unused-vars": ["error", {
+        "varsIgnorePattern": "should"
+      }]
     }  
 }
```
Comment (jcdesimp) [inline test/.eslintrc.json]: also newline here pls.

**Review by jcdesimp — CHANGES_REQUESTED** [review]
Looks good aside from the two change requests. 👍

**Comment by ShaydeNofziger** [comment]
resolves andymeneely/dev-fortress-server#2

**Review by jcdesimp — APPROVED** [review]

**Pull request merged**